src/directives/rcdni.js

```diff
@@ -0,0 +1,42 @@
+directives.directive('rcDni',['dniSemanticsValidationService','dniSyntaxValidationService',function(dniSemanticsValidationService,dniSyntaxValidationService){
+	return{
+		require: 'ngModel',					
+		link: function (scope, elm, attrs, ctrl){	
+			var validationToExecute = function (viewValue){
+				var currentTypedValue = viewValue;
+				validateDNI(currentTypedValue);
+			};
+
+			function validateDNI(dni){
+				if (validateDNISyntax(dni)){
+					validateDNISemantics(dni);
+				}
+			}
+
+			function validateDNISemantics(dni){
+				var isValid = dniSemanticsValidationService.validateDNISemantics(dni); 								
+				if (isValid){			
+					ctrl.$setValidity('pattern',true);
```

Comment:
I agree with you, I think that validateDNISyntax should set the validity of dni directive and validateDNISyntax should set the validity of pattern